### PR TITLE
Add explanation and docs link in generated tests

### LINKED ISSUE #3991: Generated Test Files: add comment with link to Redwood Testing Doc
_continuing from #3601_

There are (plenty of) cases where the boilerplate tests will fail. One recent example is #3601 re: Float and DateTime types that don't play well with Jest expect(). Instead of attempting to configure boilerplate to handle all possible situations, we decided to update the docs over here https://github.com/redwoodjs/redwoodjs.com/pull/881

But what good are docs if you don't know about them? I suggest we add a comment block to all generated test files:
- link to the Redwood Testing Doc: https://redwoodjs.com/docs/testing
- explains that boilerplate isn't intended to work for all circumstances without adjustments, e.g. Float and DateTime types

### PR DESCRIPTION
**Pull request by refactor-this**
Closes #3991.

**Added a comment to all generated test files:**
`// Generated boilerplate tests do not account for all circumstances
// and can fail without adjustments, e.g. Float and DateTime types.
//           Please refer to the RedwoodJS Testing Docs:
// https://redwoodjs.com/docs/testing#jest-expect-type-considerations
`
This should make it easier for users to debug type issues such as #3601 

Let me know if anything else should be added here @thedavidprice. Thanks!

### CODE REVIEW

**Review by thedavidprice — CHANGES_REQUESTED** [review]
@refactor-this Thanks again for moving this one forward! 

I now realize in the original issue that I did not specify _which_ test template this applies to (which was the Cell tests). You made a good judgment call applying it to all. However, this message isn't universal. But you got me thinking because each type of test template has a related Testing doc section, which it would be very helpful to link.

The message you created does apply to both of these types, which also have corresponding sections:
1. Cell tests (https://redwoodjs.com/docs/testing#testing-cells)
2. Service test (https://redwoodjs.com/docs/testing#testing-services)

In addition to only using the current copy for Cell and Service templates, what do you think about adding the respective section link as well?

For the others, we don't need the warning or link to the specific section. It would be very helpful though to link to the Testing doc in general. Could you take a stab at a concise message about Testing doc that includes a link? This will apply to:
1. component (https://redwoodjs.com/docs/testing#testing-components)
2. function (https://redwoodjs.com/docs/testing#testing-functions)
3. layout (https://redwoodjs.com/docs/testing#testing-pages-layouts)
4. page (https://redwoodjs.com/docs/testing#testing-pages-layouts)

**Comment by thedavidprice** [comment]
FYI: The CI is failing because the test snapshots need to be updated. See the failing test here:
- https://github.com/redwoodjs/redwood/runs/4903907237?check_suite_focus=true#step:11:2529

Which included the instruction about how to update locally (screenshot):
Screen Shot 2022-01-23 at 10 14 34 PM

**Comment by refactor-this** [comment]
@thedavidprice Alrighty, I went through and updated the links and text along with the snapshots 🙌

File: packages/cli/src/commands/generate/cell/templates/test.js.template [hunk 1]
```diff
@@ -2,6 +2,11 @@ import { render, screen } from '@redwoodjs/testing/web'
 import { Loading, Empty, Failure, Success } from './${pascalName}Cell'
 import { standard } from './${pascalName}Cell.mock'
 
+// Generated boilerplate tests do not account for all circumstances
+// and can fail without adjustments, e.g. Float and DateTime types.
+//           Please refer to the RedwoodJS Testing Docs:
+//        https://redwoodjs.com/docs/testing#testing-cells
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/cell/templates/test.js.template:8]:
```suggestion
//        https://redwoodjs.com/docs/testing#testing-cells
// https://redwoodjs.com/docs/testing#jest-expect-type-considerations
```

File: packages/cli/src/commands/generate/component/templates/test.tsx.template [hunk 2]
```diff
@@ -2,6 +2,9 @@ import { render } from '@redwoodjs/testing/web'
 
 import ${pascalName} from './${pascalName}'
 
+//  Please refer to the RedwoodJS Testing Docs for any issues:
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/component/templates/test.tsx.template:5]:
```suggestion
// Improve this test with help from the Redwood Testing Doc:
```

File: packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap [hunk 3]
```diff
@@ -157,6 +157,9 @@ exports[`creates a single word function file: Test snapshot 1`] = `
 
 import { handler } from './foo'
 
+//  Please refer to the RedwoodJS Testing Docs for any issues:
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap:160]:
```suggestion
//   Improve this test with help from the Redwood Testing Doc:
```

File: packages/cli/src/commands/generate/function/templates/test.ts.template [hunk 4]
```diff
@@ -2,6 +2,9 @@ import { mockHttpEvent } from '@redwoodjs/testing/api'
 
 import { handler } from './${name}'
 
+//  Please refer to the RedwoodJS Testing Docs for any issues:
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/function/templates/test.ts.template:5]:
```suggestion
//   Improve this test with help from the Redwood Testing Doc:
```

File: packages/cli/src/commands/generate/layout/__tests__/__snapshots__/layout.test.ts.snap [hunk 5]
```diff
@@ -72,6 +72,9 @@ exports[`creates a multi word layout test 1`] = `
 
 import SinglePageLayout from './SinglePageLayout'
 
+//  Please refer to the RedwoodJS Testing Docs for any issues:
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/layout/__tests__/__snapshots__/layout.test.ts.snap:75]:
```suggestion
//   Improve this test with help from the Redwood Testing Doc:
```

File: packages/cli/src/commands/generate/layout/templates/test.tsx.template [hunk 6]
```diff
@@ -2,6 +2,9 @@ import { render } from '@redwoodjs/testing/web'
 
 import ${pascalName}Layout from './${pascalName}Layout'
 
+//  Please refer to the RedwoodJS Testing Docs for any issues:
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/layout/templates/test.tsx.template:5]:
```suggestion
//   Improve this test with help from the Redwood Testing Doc:
```

File: packages/cli/src/commands/generate/page/templates/test.tsx.template [hunk 7]
```diff
@@ -2,6 +2,9 @@ import { render } from '@redwoodjs/testing/web'
 
 import ${pascalName}Page from './${pascalName}Page'
 
+//  Please refer to the RedwoodJS Testing Docs for any issues:
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/page/templates/test.tsx.template:5]:
```suggestion
//   Improve this test with help from the Redwood Testing Doc:
```

File: packages/cli/src/commands/generate/service/templates/test.ts.template [hunk 8]
```diff
@@ -15,6 +15,10 @@ const stringifyButNotScenario = (obj) => {
 import { ${pluralCamelName}<% if (crud) { %>,${singularCamelName}, create${singularPascalName}, update${singularPascalName}, delete${singularPascalName}<% } %> } from './${pluralCamelName}'
 import type { StandardScenario } from './${pluralCamelName}.scenarios'
 
+// Generated boilerplate tests do not account for all circumstances
+// and can fail without adjustments, e.g. Float and DateTime types.
+//           Please refer to the RedwoodJS Testing Docs:
+//       https://redwoodjs.com/docs/testing#testing-services
```
Comment (thedavidprice) [inline packages/cli/src/commands/generate/service/templates/test.ts.template:21]:
```suggestion
//       https://redwoodjs.com/docs/testing#testing-services
// https://redwoodjs.com/docs/testing#jest-expect-type-considerations
```

**Review by thedavidprice — APPROVED** [review]
Thank you @refactor-this (especially for being patient with me).

I wanted to make sure this is included in the release out tomorrow. Went ahead an committed tweaks I had and then updated snapshot.

Thank you so much! It's little things like this that are big wins for DX 🚀

**Comment by refactor-this** [comment]
> Thank you @refactor-this (especially for being patient with me).
> 
> I wanted to make sure this is included in the release out tomorrow. Went ahead an committed tweaks I had and then updated snapshot.
> 
> Thank you so much! It's little things like this that are big wins for DX 🚀

Awesome!!! It's my first contribution and I am very excited to have helped especially with a project I'm using 😄

**Pull request merged**